pkg/webhook/resources/node/validator.go

```diff
@@ -71,3 +95,137 @@ func validateCordonAndMaintenanceMode(oldNode, newNode *corev1.Node, nodeList []
 	}
 	return werror.NewBadRequest("can't enable maintenance mode or cordon on the last available node")
 }
+
+func (v *nodeValidator) validateCPUManagerOperation(node *corev1.Node) error {
+	annot, ok := node.Annotations[util.AnnotationCPUManagerUpdateStatus]
+	if !ok {
+		return nil
+	}
+
+	updateStatus, err := ctlnode.GetCPUManagerUpdateStatus(annot)
+	if err != nil {
+		return werror.NewBadRequest(fmt.Sprintf("Failed to retrieve cpu-manager-update-status from annotation: %v", err))
+	}
+	// only validate when update status is requested
+	if updateStatus.Status != ctlnode.CPUManagerRequestedStatus {
+		return nil
+	}
+	policy := updateStatus.Policy
+
+	// check if cpu manager policy is the same
+	if err := checkCPUManagerLabel(node, policy); err != nil {
+		return err
+	}
+	// check if there is other job that still updating cpu manager policy to the same node
+	if err := checkCurrentNodeCPUManagerJobs(node, v.jobCache); err != nil {
+		return err
+	}
+	// check if this node is master and there are other master nodes undating cpu manager policy
+	// since the policy update on master node need to restart rke2-server, we only allow one master node update policy
+	if err := checkMasterNodeJobs(node, v.nodeCache, v.jobCache); err != nil {
+		return err
+	}
+	// check if there is any vm that enable cpu pinning while cpu manager is going to be disabled
+	if err := checkCPUPinningVMIs(node, policy, v.vmiCache); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func checkCPUManagerLabel(node *corev1.Node, policy ctlnode.CPUManagerPolicy) error {
+	cpuManagerLabel, err := strconv.ParseBool(node.Labels[kubevirtv1.CPUManager])
+	// means CPUManager feature gate not enabled
+	if err != nil {
+		return werror.NewBadRequest("label cpumanager not found")
+	}
```

Comment:
This check could indicate a condition where the job is still running or has failed. 
I am not sure if this may be the best place to run this check or if may even be needed since we are checking subsequently in `checkCurrentNodeCPUManagerJobs` if the job is still running?

Reply:
Thanks for the review @ibrokethecloud , the idea of this check is to make sure harvester have kubevirt feature gate `CPUManager` enabled, in this feature gate, kubevirt update the `cpumanager` label periodically (~1 min) to check if node cpu-manager-policy is static or none. 
And in cpu_manager_controller.go#getScript https://github.com/harvester/harvester/pull/6190/files#diff-83df6178dc387757db24fc13a1b2ce905d80b0632b55bac72a63fae0b8fc590bR258-R272 I use this label to check if job succeed or fail.

---
FYI, about the kubevirt update cpumanager label,
https://github.com/kubevirt/kubevirt/blob/81e62b66e79650fbbdfa177b570ab8a4f5c5d086/pkg/virt-handler/heartbeat/heartbeat.go#L132-L157
periodically update (~1min)
https://github.com/kubevirt/kubevirt/blob/81e62b66e79650fbbdfa177b570ab8a4f5c5d086/pkg/virt-handler/heartbeat/heartbeat.go#L71-L75

Reply:
makes sense @brandboat. thanks for the clarification.